src/dmtcpworker.cpp

```diff
@@ -104,7 +104,18 @@ void restoreUserLDPRELOAD()
   //   reset LD_PRELOAD back to ENV_VAR_ORIG_LD_PRELOAD in dmtcp_nocheckpoint
   char *preload = getenv("LD_PRELOAD");
   char *userPreload = getenv(ENV_VAR_ORIG_LD_PRELOAD);
-  JASSERT(userPreload == NULL || strlen(userPreload) <= strlen(preload));
+  //   NOTE: Consider the following sequence of calls:
+  //   dmtcp_launch a.out
+  //     exec(a.out) --> this function gets called from DmtcpWorker(), and the LD_PRELOAD is destroyed
+  //       exec(b.out)
+  //    If the call to exec b.out fails, this function will be called from
+  //    dmtcpProcessFailedExec(). The following JASSERT could fail if the user
+  //    had a non-null preload env var. And so, we need to be less aggressive,
+  //    report this to the user, and let the user handle the failed exec.
+  //JASSERT(userPreload == NULL || strlen(userPreload) <= strlen(preload));
```

Comment:
@rohgarg, I'm not sure that I understand the reasoning.  The `JASSERT` was meant to protect us because 4 to 9 lines below (at lines 120-124), we will call 

`preload[0] = '\0'; ... strcat(preload, userPreload);`

We need to ensure that we don't overrun the buffer of preload.  Please see my analysis in the general discussion.